include/exadg/structure/material/material.h

```diff
@@ -41,15 +41,27 @@ class Material
   {
   }
 
+  /*
+   * Evaluate 2nd Piola-Kirchhoff stress tensor given a strain measure, e.g., the engineering
+   * strain, Green-Lagrange strain tensor or Cauchy-Green strain tensor
+   */
   virtual dealii::Tensor<2, dim, dealii::VectorizedArray<Number>>
-  evaluate_stress(dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & E,
-                  unsigned int const                                              cell,
-                  unsigned int const                                              q) const = 0;
+  second_piola_kirchhoff_stress(
+    dealii::Tensor<2, dim, dealii::VectorizedArray<Number>> const & strain_measure,
+    unsigned int const                                              cell,
+    unsigned int const                                              q) const = 0;
 
+  /*
+   * Evaluate directional derivative of the 2nd Piola-Kirchhoff stress tensor given the displacement
+   * shape function gradient Grad_delta and deformation gradient at current linearization point
```

Comment:
The above function implements the function `S(E)`, while the present function refers to the directional derivative of a function `S(d)`, with `d` being the displacement vector. Do you agree?

This is currently not reflected in the function names, but I think this would be necessary. To be more precise, maybe the documentation should say `(D S(d) / D d)|_{d_lin} * delta_d`. Would you agree to this formula?

Reply:
This is true! First point we addressed by changing the interface to the displacement gradient, which is more general than the strain measure. (For some material models, we do not really want E in the nonlinear case but actually C to compute invariants, so a common ground is Gradient(displacement)).
The second point I forgot, will adapt the name accordingly and push ASAP.